src/main/java/io/split/android/client/network/HttpStreamRequestImpl.java

```diff
@@ -1,40 +1,70 @@
 package io.split.android.client.network;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import static io.split.android.client.network.HttpRequestHelper.applySslConfig;
+import static io.split.android.client.network.HttpRequestHelper.applyTimeouts;
+import static io.split.android.client.network.HttpRequestHelper.openConnection;
+
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 
 import java.io.BufferedReader;
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.net.HttpURLConnection;
 import java.net.MalformedURLException;
 import java.net.ProtocolException;
+import java.net.Proxy;
 import java.net.URI;
 import java.net.URL;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
 
-import io.split.android.client.utils.logger.Logger;
-import okhttp3.MediaType;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.Response;
+import javax.net.ssl.SSLSocketFactory;
 
-import static com.google.common.base.Preconditions.checkNotNull;
+import io.split.android.client.utils.logger.Logger;
 
 public class HttpStreamRequestImpl implements HttpStreamRequest {
-    private static final MediaType JSON
-            = MediaType.get("application/json; charset=utf-8");
-    private OkHttpClient mOkHttpClient;
-    private URI mUri;
-    private Map<String, String> mHeaders;
-    private Response mOkHttpResponse;
-    private BufferedReader mResponseBufferedReader;
-
-    HttpStreamRequestImpl(@NonNull OkHttpClient okHttpClient, @NonNull URI uri,
-                          @NonNull Map<String, String> headers) {
-        mOkHttpClient = checkNotNull(okHttpClient);
+
+    private static final int STREAMING_READ_TIMEOUT_IN_MILLISECONDS = 80000;
```

Comment:
Perhaps putting t his constant in a "Constant class" to maintain it when changing implementation.